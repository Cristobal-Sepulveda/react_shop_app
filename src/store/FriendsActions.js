import { ADD_FRIEND, REMOVE_FRIEND } from "../utils/types";


export const addFriend = friendsIndex => (
  {
    type: ADD_FRIEND,
    payload: friendsIndex,
  }
)

export const removeFriend = friendsIndex =>(
  {
    type: REMOVE_FRIEND,
    payload: friendsIndex,
  }
)