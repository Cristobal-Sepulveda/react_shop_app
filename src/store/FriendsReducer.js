import { combineReducers } from 'redux';
import { ADD_FRIEND, REMOVE_FRIEND } from '../utils/types';
import React, {useState, useEffect} from "react";

//const [data,setData]= useState([])

const INITIAL_STATE = {current: [],
                       possible: ['Alice',
                                  'Bob',
                                  'Sammy']
                       };


const friendsReducer = (state = INITIAL_STATE, action) => {
    switch (action.type) {
        
        case ADD_FRIEND:
            // Pulls current and possible out of previous state
            // We do not want to alter state directly in case
            // another action is altering it at the same time
            const {current, possible} = state;
            // Pull friend out of friends.possible
            // Note that action.payload === friendIndex
            const addedFriend = possible.splice(action.payload, 1);

            // And put friend in friends.current
            current.push(addedFriend);

            // Finally, update the redux state
            const newState = { current, possible };
  
            return newState;

        case REMOVE_FRIEND:
            console.log(REMOVE_FRIEND)
            
        default:
            return state
  }
};

const pedidosReducer = (state = 'asd', action)=>{
    switch(action.type){
        //case x:
        default:
            return state
    }
}

export default combineReducers({friends: friendsReducer, pedidos: pedidosReducer});