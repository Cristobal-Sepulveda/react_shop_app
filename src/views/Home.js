import React, {useState, useEffect} from "react";
import { Alert, View, Text, FlatList, Button, StyleSheet, SafeAreaView, ToastAndroid} from "react-native";
import ModalPlanilla from "../modals/ModalPlanilla";
import AsyncStorage from '@react-native-async-storage/async-storage';
import { obtenerTipoConexion } from "../utils/funciones";
import { connect, useSelector } from "react-redux"
import store from "../redux/store"


const Home = ({}) =>{
  const[isRefreshing, setIsRefreshing] = useState(true)
  const[data, setData] = useState([])


  const pedidosList = useSelector(state => {
    const auxArray = []
    for (let i = 0; i< state.pedidos.allPedidos.length; i++){
        auxArray.push(JSON.stringify(state.pedidos.allPedidos[i]))
    }
    console.log("useSelector: ", auxArray)
    return auxArray
  })

    
  //funcion iniciada al hacer sync en la flatList...
  const syncFlatList = async () => {
    console.log("syncFlatList")
    setIsRefreshing(false)
    const userConexionType = await obtenerTipoConexion()
    if(userConexionType.tipoConexion == "wifi"){
      return Alert.alert("lista sincronizada")
    }
    if(userConexionType.tipoConexion == "cellular"){
      if(userConexionType.connectionDetails == "4g"){
        return Alert.alert("lista sincronizada")
      }else{
        return Alert.alert("Tu conexion de celular debe ser 4g para poder sincronizar el listado")
      }      
    }
    return Alert.alert("Debes tener conexion a internet para poder sincronizar el listado")
  }
  
  const renderItem = ({ item }) =>{
      return (<View style={styles.item}>
                <Text style={styles.title}>{item}</Text>
              </View>)
  }

  useEffect(()=>{
    setIsRefreshing(false)
  },[])


  useEffect(()=>{
    if(pedidosList.length&& !data.includes(pedidosList[(pedidosList.length-1)])){
      if(data.length === 0 && pedidosList.length>0 && !data.includes(pedidosList[(pedidosList.length-1)])){
        for(let i = 0; i<pedidosList.length;i++){
          setData(prevData =>[...prevData, pedidosList[i]])      
        }
      }else{
        setData(prevData =>[...prevData, pedidosList[(pedidosList.length-1)]])    
      }
    }
  },[pedidosList])
  
  
  const print = () => {
    console.log(data)
  }

  return (<SafeAreaView style={styles.container}>
            <Button title="loadingData" onPress={() => print()}/>
            <FlatList
              data={data}
              renderItem={renderItem}
              keyExtractor={(item) => JSON.parse(item).key}
              numColumns={1}
              backgroundColor="grey"
              ListHeaderComponent={(
                <View>
                  <Text style={styles.homeTitle}>Lista de Pedidos</Text>
                </View>)}
              ListEmptyComponent={( 
                <View>
                  <Text>
                    no hay items para desplegar
                  </Text>
                </View>)}
              refreshing = {isRefreshing}
              onRefresh={syncFlatList}
            />
            <View style={styles.buttonView}>
              <ModalPlanilla/>
            </View>
  
            <View style={styles.margenInferior}/>       

          </SafeAreaView>
         )
};


const mapStateToProps = (state) =>{
  return state
} 

const mapDispatchToProps = dispatch =>({
  addPedido: (key, nombre, rut, edad, telefono, productos, date, selectedEntrega) =>
    dispatch({
      type: Types.ADD_PEDIDO,
      payload: {key,nombre,rut,edad,telefono,productos,date,selectedEntrega}
    }),
})


const connectComponent = connect(mapStateToProps, mapDispatchToProps)
export default connectComponent(Home);


const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    item: {
        backgroundColor: 'blue',
        padding: 20,
        marginVertical: 8,
        marginHorizontal: 16,
    },
    homeTitle:{
      fontSize: 48,
      textAlign: 'center',
      color: 'white',
      backgroundColor:'black'
    },
    title: {
        fontSize: 32,
        color: 'white'
    },
    buttonView:{
      
    },
    margenInferior:{
      marginBottom: 36
    }
});

