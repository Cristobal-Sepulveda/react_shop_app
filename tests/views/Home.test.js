import React from 'react';
import Home from '../../src/views/Home';
import '@testing-library/jest-dom';
import { shallow, mount, render} from 'enzyme';
import setupTests from '../enzymeSetup/setupTests'
import { Provider}  from 'react-redux';
import store from "../../src/redux/store"
import { connect, useSelector } from "react-redux"
jest.useFakeTimers()

describe('<Home />', () => {
  
  let wrapper = render(<Provider store={store}>
                          <Home />
                        </Provider>)
  beforeEach(() => {
          wrapper = render(<Provider store={store}>
                              <Home />
                            </Provider>)
  });

  // test('Testeando el snapshot', () => {
  //    const tree = renderer.create( <Provider store={store}>
  //                                    <Home />
  //                                  </Provider>, { context: {}}).toJSON(); 
  //   const wrapper = shallow(<Home/>)
  test('Testeando el snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
});
