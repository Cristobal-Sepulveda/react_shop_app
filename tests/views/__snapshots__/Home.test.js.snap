// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Testeando el snapshot 1`] = `
<rctsafeareaview
  style="flex:1"
>
  <view
    accessibilityrole="button"
    accessibilitystate="[object Object]"
    focusable="true"
    style="opacity:1"
  >
    <view
      style="0:[object Object]"
    >
      <text
        style="0:[object Object]"
      >
        loadingData
      </text>
    </view>
  </view>
  <rctscrollview
    backgroundcolor="grey"
    data=""
    initialnumtorender="10"
    listemptycomponent="[object Object]"
    listheadercomponent="[object Object]"
    maxtorenderperbatch="10"
    refreshcontrol="[object Object]"
    scrolleventthrottle="50"
    stickyheaderindices=""
    updatecellsbatchingperiod="50"
    viewabilityconfigcallbackpairs=""
    windowsize="21"
  >
    <rctrefreshcontrol />
    <view>
      <view>
        <view>
          <text
            style="font-size:48px;text-align:center;color:white;background-color:black"
          >
            Lista de Pedidos
          </text>
        </view>
      </view>
      <view>
        <text>
          no hay items para desplegar
        </text>
      </view>
    </view>
  </rctscrollview>
  <view>
    <view
      style="flex:1;justify-content:center;align-items:center;margin-top:22px"
    >
      <modal
        animationtype="fade"
      >
        <view
          style="flex:1;justify-content:center;align-items:center;margin-top:22px"
        >
          <view
            style="margin:20px;background-color:white;border-radius:20px;padding:35px;shadow-color:#000;shadow-offset:[object Object];shadow-opacity:0.25px;shadow-radius:4px;elevation:5px"
          >
            <view>
              <text
                style="margin-top:8px;margin-bottom:4px"
              >
                Ingrese nombre
              </text>
              <textinput
                keyboardtype="default"
                style="color:#000;width:250px;height:40px;border-width:1px;border-radius:10px;padding:8px"
                underlinecolorandroid="transparent"
                value=""
              />
            </view>
            <view>
              <text
                style="margin-top:8px;margin-bottom:4px"
              >
                Ingrese rut
              </text>
              <textinput
                keyboardtype="default"
                style="color:#000;width:250px;height:40px;border-width:1px;border-radius:10px;padding:8px"
                underlinecolorandroid="transparent"
                value=""
              />
            </view>
            <view>
              <text
                style="margin-top:8px;margin-bottom:4px"
              >
                Ingrese edad
              </text>
              <textinput
                keyboardtype="numeric"
                style="color:#000;width:250px;height:40px;border-width:1px;border-radius:10px;padding:8px"
                underlinecolorandroid="transparent"
                value=""
              />
            </view>
            <view>
              <text
                style="margin-top:8px;margin-bottom:4px"
              >
                Ingrese telefono
              </text>
              <textinput
                keyboardtype="numeric"
                style="color:#000;width:250px;height:40px;border-width:1px;border-radius:10px;padding:8px"
                underlinecolorandroid="transparent"
                value=""
              />
            </view>
            <text
              style="margin-top:8px;margin-bottom:4px"
            >
              Seleccione sus pedidos
            </text>
            <view
              style="flex-direction:row"
            >
              <view
                style="margin-end:8px"
              >
                <view
                  style="flex-direction:row;margin-bottom:8px"
                >
                  <view
                    accessibilityrole="checkbox"
                    accessibilitystate="[object Object]"
                    focusable="true"
                    style="0:[object Object];2:;3:;5:"
                  />
                  <text
                    style="margin-left:8px"
                  >
                    completo
                  </text>
                </view>
                <view
                  style="flex-direction:row;margin-bottom:8px"
                >
                  <view
                    accessibilityrole="checkbox"
                    accessibilitystate="[object Object]"
                    focusable="true"
                    style="0:[object Object];2:;3:;5:"
                  />
                  <text
                    style="margin-left:8px"
                  >
                    hamburguesa
                  </text>
                </view>
              </view>
              <view>
                <view
                  style="flex-direction:row;margin-bottom:8px"
                >
                  <view
                    accessibilityrole="checkbox"
                    accessibilitystate="[object Object]"
                    focusable="true"
                    style="0:[object Object];2:;3:;5:"
                  />
                  <text
                    style="margin-left:8px"
                  >
                    jugo
                  </text>
                </view>
                <view
                  style="flex-direction:row;margin-bottom:8px"
                >
                  <view
                    accessibilityrole="checkbox"
                    accessibilitystate="[object Object]"
                    focusable="true"
                    style="0:[object Object];2:;3:;5:"
                  />
                  <text
                    style="margin-left:8px"
                  >
                    bebida
                  </text>
                </view>
              </view>
            </view>
            <view>
              <view
                style="margin-top:8px"
              >
                <view
                  accessibilityrole="button"
                  accessibilitystate="[object Object]"
                  focusable="true"
                  style="opacity:1"
                >
                  <view
                    style="0:[object Object]"
                  >
                    <text
                      style="0:[object Object]"
                    >
                      Fecha de entrega
                    </text>
                  </view>
                </view>
              </view>
            </view>
            <view>
              <text
                style="margin-top:16px"
              >
                Elija modo de entrega
              </text>
              <view
                style="border-width:1px;border-radius:10px"
              >
                <view>
                  <rncpicker
                    items="[object Object],[object Object]"
                    numberoflines="1"
                    selectedindex="0"
                    style="0:[object Object]"
                  />
                </view>
              </view>
            </view>
            <view
              style="flex-direction:row"
            >
              <view
                focusable="true"
                style="0:[object Object];1:[object Object]"
              >
                <text
                  style="color:white;text-align:center"
                >
                  Enviar Pedido
                </text>
              </view>
              <view
                focusable="true"
                style="0:[object Object];1:[object Object]"
              >
                <text
                  style="color:white;text-align:center"
                >
                  Volver
                </text>
              </view>
            </view>
          </view>
        </view>
      </modal>
      <view
        focusable="true"
        style="0:[object Object];1:[object Object]"
      >
        <text
          style="color:white;text-align:center"
        >
          Ingresar Pedido
        </text>
      </view>
    </view>
  </view>
  <view
    style="margin-bottom:36px"
  />
</rctsafeareaview>
`;
